rest_api/rest_api/pipeline/__init__.py

```diff
@@ -13,9 +13,25 @@
 
 logger = logging.getLogger(__name__)
 
-# Since each instance of FAISSDocumentStore creates an in-memory FAISS index, the Indexing & Query Pipelines would
-# end up with different indices. The same applies for InMemoryDocumentStore.
-UNSUPPORTED_DOC_STORES = (FAISSDocumentStore, InMemoryDocumentStore)
+# Each instance of FAISSDocumentStore creates an in-memory FAISS index,
+# the Indexing & Query Pipelines will end up with different indices for each worker.
+# The same applies for InMemoryDocumentStore.
+SINGLE_PROCESS_DOC_STORES = (FAISSDocumentStore, InMemoryDocumentStore)
+
+
+def _setup_indexing_pipeline(pipeline_yaml_path, indexing_pipeline_name):
```

Comment:
Since we're here, I wonder if we can make this function generic enough to be used for both query and indexing pipelines, so that the caller code would look like this:
```python
pipelines["query_pipeline"] = _load_pipeline(config.PIPELINE_YAML_PATH, config.QUERY_PIPELINE_NAME)
pipelines["indexing_pipeline"] = _load_pipeline(config.PIPELINE_YAML_PATH, config.INDEXING_PIPELINE_NAME)
```

Reply:
Done. But please review this change carefully. :grimacing: